File: src/main/java/com/mike/inject_model/resolvers/annotations/InjectModel.java
package com.mike.inject_model.resolvers.annotations;

import java.lang.annotation.*;

/**
 * Class for a search model from repository
 */
@Target(ElementType.PARAMETER)
@Inherited
@Documented
@Retention(RetentionPolicy.RUNTIME)
public @interface InjectModel {
    /**
     * @return name of path variable, if is empty use name of param
     */
    String value() default "";

    /**
     * @return if can be nullable
     */
    boolean nullable() default false;

    /**
     * @return message for not found exception
     */
    String message() default "Model [model] not found";

    /**
     * @return type of param to search
     */
    Class<?> paramType() default Long.class;

    /**
     * @return method on repository
     */
    String method() default "findById";
}

File: src/main/java/io/github/robertomike/inject_model/resolvers/annotations/InjectModel.kt
package io.github.robertomike.inject_model.resolvers.annotations

import kotlin.reflect.KClass

/**
 * Class for a search model from repository
 */
@Target(AnnotationTarget.TYPE_PARAMETER, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.TYPE)
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
annotation class InjectModel(
    /**
     * @return name of path variable, if is empty use name of param
     */
    val value: String = "",

    /**
     * @return method on repository
     */
    val method: String = "findById",

    /**
     * @return it can be nullable
     */
    val nullable: Boolean = false,

    /**
     * @return message for not found exception
     */
    val message: String = "Model [model] not found",

    /**
     * @return type of param to search
     */
    val paramType: KClass<*> = Long::class,
)
